### PR TITLE
doc: release-notes clarify about rgw encryption

### PR DESCRIPTION
**Pull request by theanalyst**
Since these will be a part of 12.2.0 major release announcements

Signed-off-by: Abhishek Lekshmanan <abhishek@suse.com>

### CODE REVIEW

File: doc/release-notes.rst [hunk 1]
```diff
@@ -42,8 +42,9 @@ Major changes from v12.0.1
   by "ceph tell mds.<id> ..."
 * RGW introduces server side encryption of uploaded objects with 3 options for
   the management of encryption keys, automatic encryption (only recommended for
-  test setups), customer provided keys similar to Amazon SSE KMS specification &
-  using a key management service (openstack barbician)
+  test setups), customer provided keys similar to Amazon SSE-C specification &
+  using a key management service (openstack barbician) similar to Amazon SSE-KMS
```
Comment (liewegas) [inline doc/release-notes.rst]: OpenStack Barbican

Comment (theanalyst) [inline doc/release-notes.rst]: ack

File: doc/release-notes.rst [hunk 2]
```diff
@@ -42,8 +42,9 @@ Major changes from v12.0.1
   by "ceph tell mds.<id> ..."
 * RGW introduces server side encryption of uploaded objects with 3 options for
   the management of encryption keys, automatic encryption (only recommended for
-  test setups), customer provided keys similar to Amazon SSE KMS specification &
-  using a key management service (openstack barbician)
+  test setups), customer provided keys similar to Amazon SSE-C specification &
```
Comment (yehudasa) [inline doc/release-notes.rst]: @theanalyst can you change `&` to `and` while you're at it?

Comment (theanalyst) [inline doc/release-notes.rst]: done

**Comment by tchaikov** [comment]
@yehudasa ping?

**Pull request merged**